gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/LiferayGradleProject.java

```diff
@@ -242,4 +243,23 @@ private GradleRunConfigurationAttributes getRunConfigurationAttributes( String t
             showExecutionView, showConsoleView );
     }
 
+    @Override
+    public <T> T adapt( Class<T> adapterType )
```

Comment:
We have two reason: 1)Because after we changed build.gradle, we need to call SynchronizeGradleProjectsJob to refresh gradle proejct for all types new ds component  , otherwise gradle project didn't recognize the new dependencies. User will have to refresh manually. We have to put the implements of AbstractLiferayModuleOperation in gradle core plugin. The project core can't invoke gradle core function. What do you means  the DSCOmponents completely inside project core? 2) Do you mean we won't support to add new maven ds component in future or maven and gradle will use same solution to add new ds component?